go-controller/pkg/ovn/egressip.go

```diff
@@ -1945,6 +1934,82 @@ func (e *egressIPZoneController) deleteEgressIPStatusSetup(name string, status e
 	return podIPs, nil
 }
 
+func (e *egressIPZoneController) ensureOnlyValidNextHops(name string, ops []libovsdb.Operation) ([]libovsdb.Operation, error) {
+	// When no nextHopIP is found, This may happen when node object is already deleted.
+	// So compare validNextHopIPs associated with current eIP.Status and Nexthops present
+	// in the LogicalRouterPolicy, then delete nexthop(s) from LogicalRouterPolicy if
+	// it doesn't match with nexthops derived from eIP.Status.
+	policyPred := func(item *nbdb.LogicalRouterPolicy) bool {
+		return item.Priority == types.EgressIPReroutePriority && item.ExternalIDs["name"] == name
+	}
+	eIP, err := e.watchFactory.GetEgressIP(name)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return ops, fmt.Errorf("error retrieving EgressIP %s object for updating logical router policy nexthops, err: %w", name, err)
+	} else if err != nil && apierrors.IsNotFound(err) {
+		// EgressIP object is not found, so delete LRP associated with it.
+		ops, err = libovsdbops.DeleteLogicalRouterPolicyWithPredicateOps(e.nbClient, ops, e.GetNetworkScopedClusterRouterName(), policyPred)
+		if err != nil {
+			return ops, fmt.Errorf("error creating ops to remove logical router policy for EgressIP %s from router %s: %v",
+				name, e.GetNetworkScopedClusterRouterName(), err)
+		}
+	} else {
+		validNextHopIPs := make(sets.Set[string])
+		for _, validStatus := range eIP.Status.Items {
+			nextHopIP, err := e.attemptToGetNextHopIP(name, validStatus)
+			if err != nil {
+				return ops, fmt.Errorf("failed to delete EgressIP %s (%s) because unable to determine next hop: %v",
+					name, validStatus.EgressIP, err)
+			}
+			validNextHopIPs.Insert(nextHopIP)
+		}
+
```

Comment:
is it possible at this point that we found no validNextHopIPs? Then should nuke the reRoutePolicies if so? Rather than just looping on valid next hops and removing invalid ones?

Reply:
yes, this may happen when none of the nodes are found on the EIP status. added code to nuke the reRoutePolicies when no validNextHopIPs are found.